keras_core/backend/jax/trainer.py

```diff
@@ -53,20 +53,27 @@ def _eager_build(self, data_batch):
             and not self._compile_metrics.built
         )
         if model_unbuilt or compile_metrics_unbuilt:
-            # Build the model on one batch of data.
+            data_spec = jax.tree_map(
```

Comment:
Please use `tree.map_structure` instead for internal consistency

Reply:
Done.

Reply:
Done.